compiler/src/dotty/tools/dotc/transform/ElimErasedValueType.scala

```diff
@@ -95,7 +96,12 @@ class ElimErasedValueType extends MiniPhase with InfoTransformer { thisPhase =>
         (sym1.owner.derivesFrom(defn.PolyFunctionClass) ||
          sym2.owner.derivesFrom(defn.PolyFunctionClass))
 
-      if (!info1.matchesLoosely(info2) && !bothPolyApply)
+      if (sym1.name != sym2.name && 
+          !(sym1.name.is(SuperAccessorName) && sym2.name.is(SuperAccessorName))
+            // super-accessors start as private, and their expanded name can clash after
+            // erasure. TODO: Verify that this is OK.
+          ||
```

Comment:
This complex condition is not very readable. Especially with the `&&` and `||` seemingly at the same level, and the large blocks. Consider:
```scala
// super-accessors start as private, and their expanded name can clash after
// erasure. TODO: Verify that this is OK.
def bothSuperAccessors =
  sym1.name.is(SuperAccessorName) && sym2.name.is(SuperAccessorName)

if ((sym1.name != sym2.name && !bothSuperAccessors) ||
    (!info1.matchesLoosely(info2) && !bothPolyApply))
  ctx.error(DoubleDefinition(sym1, sym2, root), root.sourcePos)
```